configure.ac

```diff
@@ -466,10 +466,8 @@ AC_ARG_ENABLE([coverage],
 AC_ARG_ENABLE([slow-tests],
   [AS_HELP_STRING([--enable-slow-tests],[build and run slow tests])])
 
-if test x"$with_check" = xyes; then
 AC_ARG_ENABLE([syslog-tests],
   [AS_HELP_STRING([--enable-syslog-tests],[build and run syslog tests])])
```

Comment:
Reproduced on Linux box, but had to go back in history:

```
$ make distclean
$ git co -b syslog-test  642f74d
$ ./autogen.sh && ./configure --enable-syslog-tests && make check
...
*** Warning: Linking the executable log.test against the loadable module
*** _syslog_override.so is not portable!
...
```

Hard to say what resolved this.